core/src/main/java/edu/wpi/grip/core/operations/composite/CropOperation.java

```diff
@@ -0,0 +1,123 @@
+
+
+/**
+ * Scale an image to an exact width and height using one of several interpolation modes.  Scaling
+ * images down can be a useful optimization, and scaling them up might be necessary for combining
+ * multiple images that are different sizes.
+ */
+@Description(name = "Crop Image",
+             summary = "Crop an image to an exact size",
+             category = OperationDescription.Category.IMAGE_PROCESSING,
+             iconName = "crop")
+public class CropOperation implements Operation {
+
+  private final InputSocket<Mat> inputSocket;
+  private final InputSocket<Number> xSocket;
+  private final InputSocket<Number> ySocket;
+
+  private final InputSocket<Number> widthSocket;
+  private final InputSocket<Number> heightSocket;
+  private final InputSocket<Origin> originSocket;
+
+  private final OutputSocket<Mat> outputSocket;
+
+  @Inject
+  @SuppressWarnings("JavadocMethod")
+  public CropOperation(InputSocket.Factory inputSocketFactory, OutputSocket.Factory
+      outputSocketFactory) {
+    this.inputSocket = inputSocketFactory.create(SocketHints.Inputs
+        .createMatSocketHint("Input", false));
+    this.xSocket = inputSocketFactory.create(SocketHints.Inputs
+        .createNumberSpinnerSocketHint("X", 100));
+    this.ySocket = inputSocketFactory.create(SocketHints.Inputs
+        .createNumberSpinnerSocketHint("Y", 100));
+    this.widthSocket = inputSocketFactory.create(SocketHints.Inputs
+        .createNumberSpinnerSocketHint("Width", 50));
+    this.heightSocket = inputSocketFactory.create(SocketHints.Inputs
+        .createNumberSpinnerSocketHint("Height", 50));
+    this.originSocket = inputSocketFactory
+        .create(SocketHints.createEnumSocketHint("Interpolation", Origin.CENTER));
+
+    this.outputSocket = outputSocketFactory.create(SocketHints.Outputs
+        .createMatSocketHint("Output"));
+  }
+
+  @Override
+  public List<InputSocket> getInputSockets() {
+    return ImmutableList.of(
+        inputSocket,
+        xSocket,
+        ySocket,
+        widthSocket,
+        heightSocket,
+        originSocket
+    );
+  }
+
+  @Override
+  public List<OutputSocket> getOutputSockets() {
+    return ImmutableList.of(
+        outputSocket
+    );
+  }
+
+  @Override
+  public void perform() {
+    final Mat input = inputSocket.getValue().get();
+    final Number x = xSocket.getValue().get();
+    final Number y = ySocket.getValue().get();
+    final Number width = widthSocket.getValue().get();
+    final Number height = heightSocket.getValue().get();
+
+    final Origin origin = originSocket.getValue().get();
+
+    final Rect regionOfInterest = new Rect(
+            x.intValue() + (int)(origin.xOffsetMultiplier * width.intValue()),
+            y.intValue() + (int)(origin.yOffsetMultiplier * height.intValue()),
+            width.intValue(),
+            height.intValue()
+            );
+
+    final Mat output = new Mat(input, regionOfInterest).clone();
```

Comment:
This can lead to a memory leak. @SamCarlberg Do you know of a good class to use as an example of how this should be done to avoid the memory leak?

Reply:
I don't think the clone is necessary. The  constructor used here uses the same memory as the base image IIRC so there's only minimal overhead. I'll have to take a look when I'm at my computer

Reply:
Well, we don't want this operation modifying the original input matrix.